composer/datasets/in_context_learning_evaluation.py

```diff
@@ -339,7 +349,7 @@ def split_batch(self, batch: Any, microbatch_size: int):
                 raise ValueError(f'Unexpected key {k}')
         num_chunks = len(chunked['input_ids'])
         for k, v in batch.items():
-            if isinstance(v, (int, float, str, bool, dict)):
+            if k in no_split:
```

Comment:
Just double checking, we never expect to get other keys that should not be split?

Reply:
The other keys would be handled in the above logic where we call _split_list and _default_split_batch. It's necessary that the above code correctly categorize all the keys so that they are either list, default, or no split.